File: app/src/main/java/xyz/rodit/snapmod/features/Feature.java
package xyz.rodit.snapmod.features;

public abstract class Feature extends Contextual {

    public Feature(FeatureContext context) {
        super(context);
    }

    protected void init() {

    }

    protected void onConfigLoaded(boolean first) {

    }

    protected abstract void performHooks();
}

File: app/src/main/java/xyz/rodit/snapmod/features/Feature.kt
package xyz.rodit.snapmod.features

abstract class Feature(
    context: FeatureContext, val support: LongRange = LongRange(0, Long.MAX_VALUE)
) : Contextual(context) {

    open fun init() {}
    open fun onConfigLoaded(first: Boolean) {}

    abstract fun performHooks()
}